### ISSUE TITLE
populate stock colormap sets from Eva & Nick knowledge

### ISSUE THREAD

**Issue by djhoese** [ssec/sift#23]
In GitLab by @rayg-ssec on Sep 6, 2015, 12:31

Right now Program.py only includes two color maps - grey and hot. The current setup allows for up to 16 color maps to be loaded into the GPU and easily selected among. 

Eva and Nick have color maps for a variety of purposes that could be included in the default set. Adding those to Program.py (as hardcoded data in the short term, as data files in the longer term?) would be useful until we have a colorbar editor.

**Comment by djhoese** [comment]
In GitLab by @djhoese on Sep 25, 2015, 11:56

FYI vispy includes many colormaps that can be addressed by name. A lot of the ones they've included I believe they got from matplotlib. They also have a colormap class and utility functions for creating your own. And with matplotlib's newest default colormapping change they should have utilities for generating colormaps too.

**Comment by djhoese** [comment]
In GitLab by @evas-ssec on Oct 20, 2015, 15:24

Glance has customizable colormaps (via the config inputs) but I don't have a lot of info about what people want for specific data. The Glance GUI basically has rainbow, reversed rainbow, desaturated rainbow, grayscale, and desaturated grayscale. Those are ones I picked, not something someone else asked for.

**Comment by djhoese** [comment]
In GitLab by @djhoese on Oct 20, 2015, 15:51

I can't find the other issue related to this (maybe I didn't make it), but Jordan provided us with XML copies of the colormaps that he uses most often in AWIPS. Those files were put on FTP and are probably gone now. If anyone would like to see them I think Ray and I have a copy.

...looks like I didn't actually make another issue. I had planned on converting these XML files to something vispy could actually use since there were some other vispy level things that might need to be changed. If someone else would like to take this on let me know, but for now I'll leave this issue unassigned.

**Comment by djhoese** [comment]
In GitLab by @djhoese on Oct 24, 2015, 19:55

This has been completed using vispy colormaps and their base classes. I've converted XML versions of the AWIPS colormaps provided by Jordan into vispy compatible colormaps. I was able to get clim normalization and color mapping into the fragment shader. Vispy already had colormaps done that way, but normalization was done in CPU to keep OpenGL-ES compatibility.

Issue #3 talks about having a colormap editor.
